### PR TITLE
Add Vespa Cloud Test dev steps

### PR DESCRIPTION
**Pull request by kkraune**
- a little closer - needs more work, but these steps worked for me

### CODE REVIEW

File: CI-CD/production-deployment-with-tests-java/README.md [hunk 1]
```diff
@@ -2,14 +2,50 @@
...
-    mvn test -D test.categories=system -D vespa.test.config=ext/test-config.json 
-    mvn test -D test.categories=staging-setup -D vespa.test.config=ext/test-config.json
-    mvn test -D test.categories=staging -D vespa.test.config=ext/test-config.json 
+
+## Developing system and staging tests
+Develop tests using an instance in the Dev zone.
+Use the Console and upload target/application.zip built in the steps above - use "default" instance name.
+
+    mvn test -D test.categories=system \
+             -D vespa.test.config=ext/test-config.json \
```
Comment (jonmv) [inline CI-CD/production-deployment-with-tests-java/README.md:30]:
```suggestion
```

File: CI-CD/production-deployment-with-tests-java/README.md [hunk 2]
```diff
@@ -2,14 +2,50 @@
...
+             -D dataPlaneCertificateFile=data-plane-public-cert.pem \
+             -D dataPlaneKeyFile=data-plane-private-key.pem
+
+    mvn test -D test.categories=staging-setup \
+             -D vespa.test.config=ext/test-config.json \
```
Comment (jonmv) [inline CI-CD/production-deployment-with-tests-java/README.md:35]:
```suggestion
```

File: CI-CD/production-deployment-with-tests-java/README.md [hunk 3]
```diff
@@ -2,14 +2,50 @@
...
+             -D dataPlaneCertificateFile=data-plane-public-cert.pem \
+             -D dataPlaneKeyFile=data-plane-private-key.pem
+
+    mvn test -D test.categories=staging \
+             -D vespa.test.config=ext/test-config.json \
```
Comment (jonmv) [inline CI-CD/production-deployment-with-tests-java/README.md:40]:
```suggestion
```

File: CI-CD/production-deployment-with-tests-java/ext/test-config.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
     "localEndpoints": {
-        "default" : "http://localhost:8080/"
+        "default" : "https://default.myapp.mytenant.aws-us-east-1c.dev.z.vespa-app.cloud/"
```
Comment (jonmv) [inline CI-CD/production-deployment-with-tests-java/ext/test-config.json:3]: ??

File: CI-CD/production-deployment-with-tests-java/README.md (hunk 1 above)
Comment (kkraune) [reply CI-CD/production-deployment-with-tests-java/README.md:30]: not sure I understand - you must provide the config file, right?

File: CI-CD/production-deployment-with-tests-java/ext/test-config.json (hunk 4 above)
Comment (kkraune) [reply CI-CD/production-deployment-with-tests-java/ext/test-config.json:3]: I could not find doc on the config file content - but how to provide the endpoint if not like this?

**Comment by jonmv** [comment]
The config file is only for connecting to a self-hosted container. If there's no config file, the default dev zone is used, just like you'd want. You should instead provide an example with a different dev zone, if that's relevant :)

**Comment by kkraune** [comment]
As there are multiple dev zones (AWS and GCP) I prefer to be explicit. In fact, we should ideally make this just "Endpoints" as it will be impossible to guess the right endpoint for all cases. At least the example must show use of it.

There is also no concept of a "default" Dev zone - if you upload to Dev using the console, using the default, you deploy to GCP

**Comment by jonmv** [comment]
The zones enum declared a default dev zone, and the maven plugins always respected this. 
The syntax to select one region over another is `-Dregion=...` I recently saw a ticket about choosing cloud instead, which could be relevant here.

**Comment by kkraune** [comment]
OK. but the example needs a way to deploy to the non-default regardless. As what the default really is, is undefined as of today, we might as well merge this and iterate

**Pull request merged**